frontend/src/metabase/query_builder/components/QuestionActions.unit.spec.tsx

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+import userEvent from "@testing-library/user-event";
+import { renderWithProviders, screen } from "__support__/ui";
+import { createMockState } from "metabase-types/store/mocks";
+import { createMockEntitiesState } from "__support__/store";
+import QuestionActions from "metabase/query_builder/components/QuestionActions";
+import { createMockCard, createMockNativeCard } from "metabase-types/api/mocks";
+import { getMetadata } from "metabase/selectors/metadata";
+import Question from "metabase-lib/Question";
+
+const TEST_STRUCTURED_CARD = createMockCard();
+const TEST_NATIVE_CARD = createMockNativeCard();
+
+const iconList = [
+  { label: "bookmark icon", tooltipText: "Bookmark" },
+  { label: "info icon", tooltipText: "More info" },
+  {
+    label: "Move, archive, and more...",
+    tooltipText: "Move, archive, and more...",
+  },
+];
+
+function setup({ card = TEST_STRUCTURED_CARD } = {}) {
+  const state = createMockState({
+    entities: createMockEntitiesState({
+      questions: [card],
+    }),
+  });
+
+  const metadata = getMetadata(state);
+  const question = metadata.question(card.id);
```

Comment:
nit: I think it'd make more sense to cast the `question` type here to `Question` (dropping `null`) as opposed to casting it at line 39, since it's possible that you can use `question` somewhere else and would need to cast it to `Question` again. Although it's only used once here.

```suggestion
  const question = metadata.question(card.id) as Question;
```

What do you think?

Reply:
Yeah, that makes a lot more sense